src/main/scala/com/databricks/labs/mosaic/functions/MosaicContext.scala

```diff
@@ -815,4 +818,23 @@ object MosaicContext {
 
     def reset(): Unit = instance = None
 
+    //noinspection ScalaStyle
+    def checkDBR(spark: SparkSession): Boolean = {
+        val sparkVersion = spark.conf.get("spark.databricks.clusterUsageTags.sparkVersion", "")
+        val isML = sparkVersion.contains("-ml-")
+        val isPhoton = spark.conf.get("spark.databricks.photon.enabled", "false").toBoolean
+        if (!isML && !isPhoton) {
+            // Print out the warnings both to the log and to the console
+            logWarning("DEPRECATION WARNING: Mosaic is not supported on the selected Databricks Runtime")
+            logWarning("DEPRECATION WARNING: Mosaic will stop working on this cluster from version v0.4.0+.")
+            logWarning("Please upgrade to Databricks Runtime ML or Photon-enabled Runtime.")
```

Comment:
@milos-colic Is "please upgrade to..." the right phrase or more simple "please use ..."; also, though subtle, recommend the order be (1) photon then (2) Runtime ML as we want Photon $DBUs. Suggested change:

"Please use a Databricks Photon-enabled Runtime (for performance benefits) or Runtime ML (for spatial AI benefits)."

Reply:
Sure, will update and re-push